### ISSUE TITLE
Joining multiple channels does not work

### ISSUE THREAD

**Issue by mikeputh** [JiPaix/xdccJS#500]
**Describe the bug**
Some communities require the user to join multiple channels in order to download. For example: `#MOVIEGODS` requires users to also join `#mg-chat`, otherwise the bots will simply ignore the users requests. While the CLI option `--channel [chan...]` does exist, it does not seem to function correctly. xdccJS only seems to join the first channel.

**To Reproduce**
`xdccJS --host irc.abjects.net --bot "some_bot" --download "#123" --channel "#MOVIEGODS" "#mg-chat"`

**Expected behavior**
Join both channels before DMing the bot.

**Actual behavior**
Only `#MOVIEGODS` is being joined before DMing the bot. The bot then ignores the request, because the user is not in `#mg-chat`.

**Context:**
 - OS: Linux Mint v20.3
 - Version: v4.5.0

**Comment by JiPaix** [comment]
For the sake of my own curiosity would you mind telling me what was the issue?

My first guess reading your comment was using quoting each channels instead of having both: `"#chan1" "#chan2"` instead of `"#chan1 #chan2"`

Thanks for using xdccJS 👍

**Comment by mikeputh** [comment]
My mistake was even dumber than that 😅

I did specify a rather short nickname, but did not pass the `--no-randomize` option. While observing the second channel, I simply "missed" that my user had actually joined the second channel. So, unbeknownst to me at that point, the joining of the second channel was actually working exactly as expected.

Nevertheless, I was still not getting my download started. Upon further investigation, I think that this might have something to do with the bot queuing my request, but xdccJS aborting the download after a fixed 30 second timeout. Unfortunately, the timeout is currently not configurable through the CLI (moviegods is very highly frequented and queuing times > 10min do regularly happen).

This currently leads me to a couple of "feature requests" and a bug report:
- A CLI option to configure the request timeout (to accomodate queues)
- A CLI option to log private bot messages to the console (to debug if the bot has a problem with my request or is queueing it)
- A small fix in `XdccJSbin.writeMSG` and `XdccJSbin.clearMSG` to also work in non-TTY environments (xdccJS is currently crashing when run in a non-TTY environment like `nohup xdccJS ... &` - more details [here](https://stackoverflow.com/questions/34570452/node-js-stdout-clearline-and-cursorto-functions))

Please advise me how to best proceed with these feature requests / bug fixes. I could just open issues in this repo or could build/fix them myself and open a PR.

Anyways, great project you dot going here. Absolutely love it 😃

**Comment by JiPaix** [comment]
I opened two new issues #501 and #502,


> Please advise me how to best proceed with these feature requests / bug fixes. I could just open issues in this repo or could build/fix them myself and open a PR.

If you don't feel like waiting until sunday feel free to submit a PR, i'll post details on each issues

**Comment by mikeputh** [comment]
Awesome! Thanks for the fast reply!

I probably wouldn't come around to implement/fix these myself any time soon, so I'd appreciate if you could have a look 👍

**Comment by JiPaix** [comment]
@mikeputh 
Changes are available on npm.

I was a bit lazy for the tty stuff. I basically just applied what was recommended on the stackoverflow link without much testing.
feel free to open a new issue or [come by discord](https://discord.gg/x6VssfkW) if you have any questions.

documentation for queue handling is available [here](https://github.com/JiPaix/xdccJS#usage)

**Comment by mikeputh** [comment]
Wow, that was fast. Thanks a lot!
- The non-TTY fix is working nicely. The crashing issue is fixed. Only the progress bar doesn't show up anymore, but that's fine.
- The bot messages are really helpful. I did not encounter a queue yet, but will report back as soon as I get the chance to try it out. Nevertheless, I have encountered very slow responding bots that did neither announce a queue, nor started the transfer within the 30s. A configurable timeout option for these cases would still be superb.
- I noticed a problem, that my downloads keep getting aborted at random points now, which I have not encountered before. I am still investigating if it's due to a shaky internet connection or if the download gets aborted when a Bot message comes in (e.g. new pack announcement). I suspect the latter, but don't know for sure.

Log:
```
> connected to : <NETWORK>
  > joined: [ #CHANNEL1 #CHANNEL2 ]
    > sending command: /MSG <BOT NAME> xdcc send <PACKAGE NUMBER>
        @<BOT NAME>: ** Sending you pack #<PACKAGE NUMBER> ("<PACKAGE NAME>") ...
     [i] downloading : <PACKAGE NAME>
      | [=============       ] ETA: 3 minutes 8s @ 14.98 MB/s - 68%[X] Connection error: read ECONNRESET
     [X] Connection error: read ECONNRESET
      [i] retrying: 1/1
[X] Connection error: read ECONNRESET
     [X] Connection error: read ECONNRESET
      [X] skipped pack: <PACKAGE NUMBER>
      [X] couldn't download pack: <PACKAGE NUMBER> from <BOT NAME>
```

**Comment by JiPaix** [comment]
fix (i think) available in v`4.5.28`
